src/calibrationlib/operator_feature_export.hpp

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <string>
+
+#include <operators/abstract_operator.hpp>
+#include "csv_writer.hpp"
+
+namespace opossum {
+
+class OperatorFeatureExport {
+ public:
+  explicit OperatorFeatureExport(const std::string& path_to_dir);
+
+  void export_to_csv(std::shared_ptr<const AbstractOperator> op) const;
+
+ private:
+  const std::string& _path_to_dir;
+
+  std::optional<const std::vector<std::string>> _get_header(const OperatorType type) const {
+    std::stringstream ss;
+    switch (type) {
+      case OperatorType::TableScan:
+        return std::vector<std::string>({"INPUT_ROWS_LEFT", "OUTPUT_ROWS", "RUNTIME_NS", "SCAN_TYPE", "TABLE_NAME",
+                                         "COLUMN_NAME", "SCAN_IMPLEMENTATION"});
+      default:
+        return {};
```

Comment:
`Fail(...)`?

Reply:
Ah, now that I read more of that code, I see that this would not work because you rely on the optional being returned here.